### ISSUE TITLE
Extraction fails with `select_trait_candidate`

### ISSUE THREAD

**Issue by franziskuskiefer** [hacspec/hax#461]
A bunch of functions in Bertie fail with `select_trait_candidate`. They all seem to use `collect`.

Here are tests for them.
https://github.com/hacspec/hax/pull/new/franziskus/select_trait_candidate

**Comment by franziskuskiefer** [comment]
Looks like the fix in #478 has been incomplete.

hax crashes with

```
thread 'rustc' panicked at '`ProjectionPredicate(AliasTy { substs: [P, (&<I as std::iter::Iterator>::Item,)], def_id: DefId(2:2931 ~ core[02ea]::ops::function::FnOnce::Output) }, Term::Ty(bool))` has escaping bound vars, so it cannot be wrapped in a dummy binder.', /rustc/d59363ad0b6391b7fc5bbb02c9ccf9300eef3753/compiler/rustc_middle/src/ty/mod.rs:1195:26
```

on the following code
```rust
fn iterating(list: Vec<u8>) {
    let _indices: Vec<_> = (0..5).filter(|i| list.iter().any(|n| n == i)).collect();
}
```

**Comment by W95Psp** [comment]
Closing this in favor of #495, the issues in https://github.com/hacspec/hax/pull/new/franziskus/select_trait_candidate are gone
